src/libmongoc/src/mongoc/mongoc-server-description-private.h

```diff
@@ -183,4 +184,11 @@ void
 mongoc_server_description_set_topology_version (mongoc_server_description_t *sd,
                                                 const bson_t *tv);
 
+/* If a service_id is set on the topology description, copies it to @oid and
+ * returns true. Otherwise returns false and zeros out oid.
+ */
+bool
+mongoc_server_description_service_id (
+   const mongoc_server_description_t *description, bson_oid_t *oid);
```

Comment:
nit: It's probably overly-pedantic, as basically nobody writes this, but did you mean:

bool
mongoc_server_description_service_id (
   const mongoc_server_description_t * const description, bson_oid_t *oid);

Reply:
I did not. I do not think a const pointer adds a guarantee from the caller's perspective.

If `mongoc_server_description_service_id` promises not to change what description points to, a caller could not observe it.

Reply:
+1. And I don't think it really buys anything, either.